src/Plugin/Block/CorporateFormBlock.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_contact_forms\Plugin\Block;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\contact\Entity\ContactForm;
+use Drupal\Core\Entity\EntityFormBuilderInterface;
+
+/**
+ * Provides a block that renders a corporate form.
+ *
+ * @Block(
+ *   id = "oe_contact_forms_corporate_block",
+ *   admin_label = @Translation("Corporate form"),
+ *   category = @Translation("Corporate forms"),
+ *   deriver = "Drupal\oe_contact_forms\Plugin\Derivative\CorporateFormBlock",
+ * )
+ */
+class CorporateFormBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The EntityFormBuilder.
+   *
+   * @var \Drupal\Core\Entity\EntityFormBuilder
+   */
+  protected $entityFormBuilder;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new CorporateFormBlock.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param array $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityFormBuilderInterface $entity_form_builder
+   *   The entity form builder.
+   */
+  public function __construct(array $configuration, string $plugin_id, array $plugin_definition, EntityTypeManagerInterface $entityTypeManager, EntityFormBuilderInterface $entity_form_builder) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->entityFormBuilder = $entity_form_builder;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('entity.form_builder')
+    );
+  }
+
+  /**
+   * When the contact form is modified, these cache tags will be invalidated.
+   */
+  public function getCacheTags() {
```

Comment:
Instead of using this method and having to load the contact form twice (once here and once in the build), you can just add all the cache metadata in the buildForm build..

Reply:
refactored